ui/app/components/account-menu/index.js

```diff
@@ -103,6 +118,54 @@ AccountMenu.prototype.render = function () {
   ])
 }
 
+
+/**
+ * Renders the Account-New Menu-Entries dynamically, usin "keyrings" data
+ * 
+ * Core provides all relevant data within keyrings[] (see ... )
+ * 
+ * TODO: make the relevant data available in keyrings (or similar state object), then use here
+ */
+AccountMenu.prototype.renderAccountNew = function () {
+
+  // Keyrings should hold: createAccount/importAccount, img.account-menu__item-icon, 
+  // images/import-account.svg, UI type(AccountPage/ ?) UI subtype(CREATE/IMPORT)
+  // UI type will be later supplemented with e.g. "HardWalletPage"
+
+  // For now, please just provide a basic working implementation, thus the next
+  // steps are enabled. (a few days more patience)
+
+  const {
+    keyrings,
+  } = this.props
+
+  // ideally, keyrings would be an array of objects, which hold all the relevant data
+  // to avoid breaking existent behaviour, a new keyrings-data could(!) be introduced
+  var temp = [];
+  for (var i=0; i <= keyrings.length ; i++) {
+    temp.push(
+      h(Item, {
+        // use "this.props.", otherwise event is not fired
+        onClick: () => this.props.showNewAccountPage('CREATE'),
```

Comment:
If you include `showNewAccountPage` in the destructured assignment from `this.props` above, you don't need to include `this.props` here.

So, line 138 - 141 would become:
```
const {
    keyrings,
    showNewAccountPage,
} = this.props
```

Reply:
Ahh, there it is, because this is its own function, with its own scope. I missed this because he does the destructuring correctly [here](https://github.com/MetaMask/metamask-extension/pull/3667/files#diff-d2dbe35a9628ad9385007554ef0ca067R85).

Reply:
@danjm , @danfinlay - All good. "this.props" and "destructuring" are now friends of mine!